services/src/main/java/org/exoplatform/addons/gamification/entities/domain/effective/GamificationActionsHistory.java

```diff
@@ -168,8 +168,34 @@
         + " AND g.date BETWEEN :fromDate AND :toDate"
         + " AND ((g.ruleId IS NOT NULL AND g.ruleId IN (:ruleIds)) \n"
         + "      OR (g.actionTitle IS NOT NULL AND g.actionTitle IN (:ruleEventNames))) \n"
-        + " ORDER BY g.createdDate DESC"
+        + " ORDER BY g.id DESC"
 )
+@NamedQuery(
+    name = "GamificationActionsHistory.findRealizationsByDateDescendingUserAccessPrivilege",
+    query = "SELECT DISTINCT g FROM GamificationActionsHistory g "
+        + " WHERE g.earnerType = :type"
+        + " AND g.earnerId = :earnerId"
+        + " AND g.date BETWEEN :fromDate AND :toDate"
+        + " ORDER BY g.id DESC"
+)
+@NamedQuery(
+    name = "GamificationActionsHistory.findRealizationsByDateAscendingUserAccessPrivilege",
+    query = "SELECT DISTINCT g FROM GamificationActionsHistory g"
+        + " WHERE g.earnerType = :type"
+        + " AND g.earnerId = :earnerId"
+        + " AND g.date BETWEEN :fromDate AND :toDate"
+        + " ORDER BY g.id ASC"
+)
+@NamedQuery(
+    name = "GamificationActionsHistory.findRealizationsByDateAndRulesUserAccessPrivilege",
```

Comment:
Useless query ?

Reply:
actually this query has been named to ByEarnerAndDateAndRules
fixed in the next commit

Reply:
wrong commit cherry-picked actually . 
it should be fixed by now